### DISCUSSION TITLE
Help with recovering deleted files

### DISCUSSION THREAD

**Discussion by balamu783** [community/community#22331] [New to GitHub]
<p>I require support for the issue I am facing, below says the problem I am facing,</p>
<ol>
<li>My internet connection is disabled.</li>
<li>Right clicked in my desktop and selected "Create Repository"option using Git</li>
<li>Once repository is created, then all the files in the desktop got a ‘blue colored question mark’ icon. I never committed the files. </li>
<li>I used Git Clean using the UI to remove the reference, just right clicked in the desktop and choose Tortoise Git -&gt; Clean -&gt; only selected ‘Do not use Recycle Bin’ option.</li>
<li>Now Git clean command ran and all my files in the desktop were gone.</li>
<li>Then I ran the $ git fsck  --lost-found</li>
<li>I got some files inside the folder named as ‘lost-found’</li>
<li>All the file names were hashed with SHA1 &amp; file types were missing.</li>
</ol>
<p>Please help me in getting the deleted files, since all were related to my project and I accidently deleted them.</p>
<p>Kindly help me in recovering my files.</p>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/balamu783">@balamu783</a>,</p>
<p>Welcome to the Community! You asked a new question in the comments of a not-completely related thread that was already marked as  <strong>Resolved</strong>. Please note that your question may get more engagement if you just open up a new topic in the correct board.</p>

**Comment by nadiajoyce** [comment]
<p>Thank you <a class="mention" href="/mpboom">@mpboom</a>. That is correct.</p>
<p><a class="mention" href="/balamu783">@balamu783</a>, thanks for being here! I’ve moved your comment from an already-existing thread into it’s own topic. Because your question is slightly different from the original thread you posted in, you’ll more likely get help with this open as a new topic.</p>
<p>Best of luck!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Unfortunately, this sounds like you’re asking a question about <a href="https://tortoisegit.org" rel="nofollow noopener">TortoiseGit</a> and I’m not sure how their Clean functionality works. You may want to check with the <a href="https://tortoisegit.org/support/" rel="nofollow noopener">support page for TortoiseGit itself</a>. If it uses the underlying <a href="https://git-scm.com/docs/git-clean" rel="nofollow noopener">git clean</a> implementation, that is a command specifically for deleting files that are not included in your Git repository, meaning that there is no record or backup of them in Git. Unfortunately, this most likely means that unless they went to some sort of OS-specific recycle bin or trash folder, that the files are not going to be easily recoverable and not something that we’re going to be able to assist with.</p>
<p>Let us know if you have more questions.</p>

**Comment by balamu783** [comment]
<p>Nope :frowning:</p>
<p>I have selected the ‘Do not use recycle bin’ option and the files got bypassed and deleted permanently (Shift + Delete).</p>
<p>But I see some of the files (hashed) inside the ‘.git -&gt; objects’ folder location. Looking for a way to recover those files using any Git Command.</p>

**Comment by lee-dohm** [comment]
<p>If the files were tracked by Git, then you should be able to use the <a href="https://git-scm.com/docs/git-reflog" rel="nofollow noopener"><code>git reflog</code> feature</a> to go backwards in time and restore your repository to a previous state. Check out <a href="https://github.blog/2015-06-08-how-to-undo-almost-anything-with-git/" rel="nofollow noopener">this GitHub blog post</a> for information on the feature and some sample scenarios. However, if the files were not tracked by Git (because <code>git clean</code> only deletes files not tracked by Git), then that feature may still not help.</p>

**Comment by 2000calories** [comment]
@balamu783 Did you find a way to recover your data? same issue here. haha
